src/Microsoft.Azure.Functions.Extensions.Mcp/Http/StreamableHttpRequestHandler.cs

```diff
@@ -2,18 +2,214 @@
+        {
+            return;
+        }
+
+        var session = await GetOrCreateSessionAsync(context, mcpOptions.Value);
+
+        if (session is null)
+        {
+            return;
+        }
+
+        try
+        {
+            McpHttpUtility.SetSseContext(context);
+
+            var responseWritten = await session.Transport.HandlePostRequest(new HttpDuplexPipe(context), context.RequestAborted);
+
+            if (!responseWritten)
+            {
+                context.Response.Headers.ContentType = (string?)null;
+                context.Response.StatusCode = StatusCodes.Status202Accepted;
+            }
+        }
+        finally
+        {
+            if (session.Transport.IsStateless)
+            {
+                // In stateless mode, we do not store the session in the session manager.
+                // The session is recreated for each request based on the MCP-Session-Id.
+                // The session ID is written to the response header in the transport.
+                await session.DisposeAsync();
+            }
+        }
+    }
+
+    private async Task<bool> PostHasValidMediaTypesAsync(HttpContext context)
+    {
+        // As per the MCP spec, ensure that, when using streamable HTTP, the client accepts both
+        // application/json and text/event-stream media types.
+        var typedHeaders = context.Request.GetTypedHeaders();
+        if (!typedHeaders.Accept.Any(h => h.MatchesMediaType("application/json"))
+            || !typedHeaders.Accept.Any(h => h.MatchesMediaType("text/event-stream")))
+        {
+            await WriteJsonRpcErrorAsync(context,
+                "Not Acceptable: Client must accept both application/json and text/event-stream",
+                StatusCodes.Status406NotAcceptable);
+
+            return false;
+        }
+
+        return true;
+    }
+
+    private async ValueTask<IMcpClientSession<StreamableHttpTransport>?> GetOrCreateSessionAsync(HttpContext context, McpOptions mcpOptionsValue, bool stateless = true)
+    {
+        if (!stateless)
+        {
+            throw new NotSupportedException("Stateful sessions are not supported.");
+        }
+
+        var sessionId = context.Request.Headers[McpSessionIdHeaderName].ToString();
+
+        if (string.IsNullOrEmpty(sessionId))
+        {
+            return await StartNewSessionAsync(context, stateless);
+        }
+
+        if (!ClientStateManager.TryParseUriState(sessionId, out string? clientId, out string? instanceId, mcpOptionsValue.EncryptClientState))
+        {
+            await WriteJsonRpcErrorAsync(context, "Invalid client state.");
+
+            return null;
+        }
+
+        return await RebuildStatelessSessionAsync(context);
+    }
+
+    private async ValueTask<IMcpClientSession<StreamableHttpTransport>> StartNewSessionAsync(HttpContext context, bool stateless)
+    {
+        if (!stateless)
+        {
+            throw new NotSupportedException("Stateful mode is not supported.");
```

Comment:
Can we make these unsupported messages consistent? Above on line 100 we have `Stateful sessions are not supported.`